luaui/Widgets/cmd_area_commands_filter.lua

```diff
@@ -0,0 +1,489 @@
+	return {
+		name = "Area Command Filter",
+		desc = "Hold Alt or Ctrl with an area command (Reclaim, Load, Attack, etc.) centered on a unit or feature to filter targets.",
+		author = "SuperKitowiec. Based on Specific Unit Reclaimer and Loader by Google Frog",
+		date = "October 16, 2025",
+		license = "GNU GPL, v2 or later",
+		layer = 0,
+		enabled = true
+	}
+end
+
+local spGiveOrderToUnitArray = Spring.GiveOrderToUnitArray
+local spGetSelectedUnits = Spring.GetSelectedUnits
+local spGetUnitsInCylinder = Spring.GetUnitsInCylinder
+local spWorldToScreenCoords = Spring.WorldToScreenCoords
+local spTraceScreenRay = Spring.TraceScreenRay
+local spGetUnitDefID = Spring.GetUnitDefID
+local spGetUnitAllyTeam = Spring.GetUnitAllyTeam
+local spGetFeatureDefID = Spring.GetFeatureDefID
+local spGetFeaturesInCylinder = Spring.GetFeaturesInCylinder
+local spGetSpectatingState = Spring.GetSpectatingState
+local spGetGameFrame = Spring.GetGameFrame
+local spGetMyTeamID = Spring.GetMyTeamID
+local spGetMyAllyTeamID = Spring.GetMyAllyTeamID
+local spIsReplay = Spring.IsReplay
+local spGetUnitIsTransporting = Spring.GetUnitIsTransporting
+local spGetUnitPosition = Spring.GetUnitPosition
+local spGetFeaturePosition = Spring.GetFeaturePosition
+local spGetUnitArrayCentroid = Spring.GetUnitArrayCentroid
+local spGetFeatureResurrect = Spring.GetFeatureResurrect
+
+local myTeamID
+local myAllyTeamID
+local gameStarted
+
+---------------------------------------------------------------------------------------
+--- Target sorting logic (pick the closest first)
+---------------------------------------------------------------------------------------
+
+---@field position1 table {x, y, z}
+---@field position2 table {x, y, z}
+local function distanceSq(position1, position2)
+	local dx = position1.x - position2.x
+	local dz = position1.z - position2.z
+	return dx * dx + dz * dz
+end
+
+---@return table {x, y, z}
+local function toPositionTable(x, y, z)
+	return { x = x, y = y, z = z }
+end
+
+----------------------------------------------------------------------------------------------------------
+--- Logic which distributes targets between transports. Should be split and extracted to separate widget
+--- Preferably after https://github.com/beyond-all-reason/Beyond-All-Reason/pull/5738 will be merged
+----------------------------------------------------------------------------------------------------------
+
+-- Multiplier to convert footprints sizes
+-- see SPRING_FOOTPRINT_SCALE in GlobalConstants.h in recoil engine repo for details
+-- https://github.com/beyond-all-reason/RecoilEngine/blob/master/rts%2FSim%2FMisc%2FGlobalConstants.h
+local springFootprintScale = 2
+
+local transportDefs = {}
+local cantBeTransported = {}
+local unitMass = {}
+local unitXSize = {}
+
+for defId, def in pairs(UnitDefs) do
+	if def.transportSize and def.transportSize > 0 then
+		transportDefs[defId] = { def.transportMass, def.transportCapacity, def.transportSize, def.health }
```

Comment:
consider named members (`mass = def.transportMass, capacity = def.transportCapacity` etc), since `def[4]` is a bit magic and this table won't likely be large enough to warrant optimizing it into an array